iOS: Improve accessibility of multiline TextInputs

rigdern opened the pull request:
Fixes #13691.

Prior to this change, VoiceOver generally didn't say anything when selecting a multiline TextInput. The only exception was VoiceOver would announce the TextInput's placeholder if it had one set.

After this change, VoiceOver announces the following for multiline TextInputs:
  - The TextInput's content.
  - The TextInput's placeholder if the TextInput doesn't have any content in it.
  - The fact that the TextInput is a textfield.
  - The cursor position if the TextInput is being edited.

This is similar to the behavior of single line TextInputs.

This change achieves this by disabling `RCTTextView` as an accessibility element. `RCTTextView` is a subclass of `RCTView` so VoiceOver doesn't recognize this as a textfield. Instead, VoiceOver now sees the child `RCTUITextView` which is a subclass of `UITextView` so VoiceOver does recognize it as a textfield. Additionally, an `accessibilityLabel` implementation was added to `RCTUITextView` in order to take the value of the placeholder into account.

## Test Plan (required)

Verified the announcements of TextInputs with various props:
  - No placeholder and no content
  - Just a placeholder
  - Just content
  - Both a placeholder and content

Did this for both singe line inputs and multiline inputs. For setting content in multiline inputs, I tested both using the `value` prop and passing children. All other props being equal, these configurations resulted in similar announcements.

I verified that the following accessibility props work the same on singleline and multiline TextInputs:
  - `accessible`
  - `accessibilityLabel`
  - `accessibilityTraits`
  - `accessibilityViewIsModal`
  - `onAccessibilityTap`
  - `onMagicTap`

Additionally, my team has been using this change in our app.

Adam Comella
Microsoft Corp.

shergin commented on Libraries/Text/RCTUITextView.m:
```diff
@@ -51,6 +51,25 @@ - (void)dealloc
 
+- (NSString *)accessibilityLabel
+{
+  NSMutableString *accLabel = [NSMutableString stringWithString:@""];
```
Should be `NSMutableString *accessibilityLabel = [NSMutableString new];`.
(Please, use `new` and, please, avoid abbreviation.)

shergin commented:
(Just for reference: This PR also fixes https://github.com/facebook/react-native/issues/13691 and internal task t17745139.)

shergin commented:
And seems that this PR also breaks common support for `accessibilityLabel` and co.

rigdern commented:
@shergin Can you elaborate on what this PR breaks?

rigdern commented:
@shergin I see what you mean. It was confusing because `accessibilityLabel` is both the name of a method I implemented and the name of a prop. Initially, I didn't realize you were talking about the prop.

I addressed your feedback and ensured that the accessibility props work the same on singleline and multiline TextInputs:
  - `accessible`
  - `accessibilityLabel`
  - `accessibilityTraits`
  - `accessibilityViewIsModal`
  - `onAccessibilityTap`
  - `onMagicTap`

shergin commented:
@rigdern Sorry, I was not clear! 😞 
I am really appreciate your effort. Accessibility stuff is really important!
Let me think about it couple of days and we will definitely merge this next week.

shergin commented:
@rigdern I just landed 072d2709df5ca866a300f58ab4d2934aa0fdde66 which removes necessity to proxy all accessibility props to subview. So, we have to revert all changes in `RCTTextView.m`. Sorry about that.

shergin commented:
I just did it. I believe we are good to go! I will merge it on Monday.
